### PR TITLE
feat: make log format configurable

### PR DESCRIPTION
**Pull request by Kraego**
Since I had problems with the encoding of the colored json log in the OKD shell (see Image), I've made the logformat configurable.

image

### CODE REVIEW

**Comment by Kraego** [comment]
Would it be better to use `LANGFLOW_LOG_ENV`? Just found the variable in the code, currently there is no decription in the environments_variables.md for this setting.

**Comment by ogabrielluiz** [comment]
Hey @Kraego 

How are you? 

This is a valid problem. If we were to use LANGFLOW_ENV we'd have to build something like the following to check for valid format strings:
```python
import logging

def is_valid_log_format(format_string):
    """Validates a logging format string by attempting to format it with a dummy LogRecord.

    Args:
        format_string (str): The format string to validate.

    Returns:
        bool: True if the format string is valid, False otherwise.
        str: Error message if invalid, empty string if valid.
    """
    record = logging.LogRecord(
        name="dummy",
        level=logging.INFO,
        pathname="dummy_path",
        lineno=0,
        msg="dummy message",
        args=None,
        exc_info=None
    )

    formatter = logging.Formatter(format_string)

    try:
        # Attempt to format the record
        formatter.format(record)
    except (KeyError, ValueError, TypeError) as e:
        # Formatting failed, return False and the error message
        msg = f"Invalid log format string: {format_string}"
        raise ValueError(msg) from e
    return True

```

What do you think?

**Review by ogabrielluiz — APPROVED** [review]
Thanks @Kraego !

LGTM

**Comment by Kraego** [comment]
@ogabrielluiz now ready for merge

**Comment by ogabrielluiz** [comment]
Hey @Kraego 

Sounds good! 

I'll merge it ASAP. You can leave it out of date because it will only merge when no other PR is merging.

File: src/backend/base/langflow/logging/logger.py [hunk 1]
```diff
@@ -166,12 +171,36 @@ async def write_async(self, message):
...
+        name="dummy", level=logging.INFO, pathname="dummy_path", lineno=0, msg="dummy message", args=None, exc_info=None
+    )
+
+    formatter = logging.Formatter(format_string)
+
+    try:
+        # Attempt to format the record
+        formatter.format(record)
+    except (KeyError, ValueError, TypeError):
+        return False
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/logging/logger.py:193]: I feel like we should log an error here so the user know why their log format isn't working.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

**Review by ogabrielluiz — APPROVED** [review]

**Pull request merged**